src/main/java/com/archisacadeny/student/StudentRepository.java

```diff
@@ -65,4 +67,65 @@ public Student createStudent(Student student) {
 
         return student;
     }
+
+    public StudentReport generateStudentAchievementReport(int studentId) {
+        String query = """
+        SELECT s.Id, s.FullName, c.Id AS CourseId, c.CourseName, e.Grade
+        FROM students s
+        JOIN enrolled_courses e ON s.Id = e.studentId
+        JOIN courses c ON e.courseId = c.Id
+        WHERE s.Id = ?
+        """;
+
+        try (PreparedStatement statement = DataBaseConnectorConfig.getConnection().prepareStatement(query)) {
+            statement.setInt(1, studentId);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (!resultSet.next()) {
+                    throw new IllegalArgumentException("Student with ID " + studentId + " not found.");
+                }
+
+                long studentIdFromDb = resultSet.getLong("Id");
+                String studentName = resultSet.getString("FullName");
+
+                List<CourseGrade> courseGrades = new ArrayList<>();
+                int totalGradePoints = 0;
+                int totalCredits = 0;
+                double gpa = 0.0;
+
+                do {
```

Comment:
you don't need to create a sperate table for storing grades and etc it is already there in Oruj's PR please do take a look together